model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py

```diff
@@ -507,12 +507,30 @@ async def execute(
         quantize: Optional[Quantization],
         checkpoint_path: Optional[str],
         chat_template_override: Optional[str],
+        nodes_per_worker: int,
     ) -> ModelBundle:
+        multinode = nodes_per_worker > 1
         if source == LLMSource.HUGGING_FACE:
             self.check_docker_image_exists_for_image_tag(
                 framework_image_tag, INFERENCE_FRAMEWORK_REPOSITORY[framework]
             )
-            if framework == LLMInferenceFramework.DEEPSPEED:
+            if multinode and framework == LLMInferenceFramework.VLLM:
+                bundle_id = await self.create_vllm_multinode_bundle(
```

Comment:
nit: maybe move this into the framework == LLMInferenceFramework.VLLM block instead?